File: ru-snoozing/src/tts.js
import fs from 'fs';
import 'dotenv/config';
import axios from 'axios';
import player from 'play-sound';
import { promisify } from 'util';

const apiKey = process.env.ELEVENLABS_API_KEY;
const voiceId = 'TX3LPaxmHKxFdv7VOQHJ';

console.log("🎤 TTS script started");
console.log("API Key present:", !!apiKey);

// Get text from command-line argument
const text = process.argv[2];

if (!text) {
  console.error("❌ No text provided as argument");
  process.exit(1);
}

console.log("📝 Text to convert:", text);

async function main() {
  try {
    // Generate speech
    console.log("🔄 Calling ElevenLabs API...");
    const response = await axios.post(
      `https://api.elevenlabs.io/v1/text-to-speech/${voiceId}`,
      {
        text,
        model_id: 'eleven_monolingual_v1',
      },
      {
        headers: {
          'xi-api-key': apiKey,
          'Content-Type': 'application/json',
        },
        responseType: 'arraybuffer',
      }
    );

    // Save audio file
    const outputPath = 'output.mp3';
    fs.writeFileSync(outputPath, response.data);
    console.log('✅ Audio file saved as output.mp3');

    // Play audio and wait for completion
    console.log('🔊 Starting audio playback...');
    const audioPlayer = player({});
    
    // Promisify the play function to use async/await
    const playAudio = promisify(audioPlayer.play.bind(audioPlayer));
    
    try {
      await playAudio(outputPath);
      console.log('✅ Audio playback completed');
    } catch (playErr) {
      console.error('❌ Error playing audio:', playErr);
      process.exit(1);
    }

    // Clean exit
    process.exit(0);

  } catch (err) {
    console.error('❌ Error generating speech:');
    if (err.response) {
      console.error('Status:', err.response.status);
      console.error('Data:', err.response.data?.toString());
    } else {
      console.error(err.message);
    }
    process.exit(1);
  }
}

main();

File: ru-snoozing/backend/tts.py
import os
import requests
from dotenv import load_dotenv
from playsound import playsound

load_dotenv()
api_key = os.getenv("ELEVENLABS_API_KEY")
voice_id = "TX3LPaxmHKxFdv7VOQHJ"

def text_to_speech(text: str) -> str:
    """
    Generate speech from text using ElevenLabs API.
    Saves to output.mp3 and plays it.
    Returns the file path of the saved MP3.
    """
    try:
        url = f"https://api.elevenlabs.io/v1/text-to-speech/{voice_id}"
        headers = {
            "xi-api-key": api_key,
            "Content-Type": "application/json",
        }
        payload = {
            "text": text,
            "model_id": "eleven_monolingual_v1"
        }

        response = requests.post(url, headers=headers, json=payload)

        if response.status_code != 200:
            print("❌ Error generating speech:", response.text)
            return ""

        # Save to backend/output.mp3
        output_file = os.path.join(os.path.dirname(__file__), "output.mp3")
        with open(output_file, "wb") as f:
            f.write(response.content)

        print("✅ Audio file saved as output.mp3")

        # Play audio locally
        playsound(output_file)
        print("🔊 Playing audio...")

        return output_file

    except Exception as e:
        print("❌ Error:", e)
        return ""
